sdk/python/feast/infra/registry/remote.py

```diff
@@ -39,8 +41,16 @@ def __init__(
         registry_config: Union[RegistryConfig, RemoteRegistryConfig],
         project: str,
         repo_path: Optional[Path],
+        auth_config: AuthConfig,
     ):
-        self.channel = grpc.insecure_channel(registry_config.path)
+        if auth_config.type in ["oidc", "kubernetes"]:
```

Comment:
Can we move this logic to util method so that this auth logic is abstracted?

Reply:
Pls use a util function in `auth_client_manager` module to do the job, like:
`def init_grpc_auth_client_manager(auth_config: AuthConfig): ...`

BTW: anyway pls consider also my next comment ...